src/timedisc/timedisc.jl

```diff
@@ -156,6 +156,167 @@ function Base.resize!(integrator::SimpleIntegrator2N, new_size)
+    end
+  elseif !isnothing(callback)
+    error("unsupported")
+  end
+
+  solve!(integrator)
+end
+
+function solve!(integrator::SimpleIntegrator3Sstar)
+  @unpack prob = integrator.sol
+  @unpack alg = integrator
+  t_end = last(prob.tspan)
+  callbacks = integrator.opts.callback
+
+  integrator.finalstep = false
+  @timeit_debug timer() "main loop" while !integrator.finalstep
+    if isnan(integrator.dt)
+      error("time step size `dt` is NaN")
+    end
+
+    # if the next iteration would push the simulation beyond the end time, set dt accordingly
+    if integrator.t + integrator.dt > t_end || isapprox(integrator.t + integrator.dt, t_end)
+      integrator.dt = t_end - integrator.t
+      terminate!(integrator)
+    end
+
+    # one time step
+    integrator.u_tmp .= zero(eltype(integrator.u_tmp))
+    integrator.u_tmp2 .= zero(eltype(integrator.u_tmp2))
+    integrator.u_tmp3 .= integrator.u
+    for stage in eachindex(alg.c)
+      t_stage = integrator.t + integrator.dt * alg.c[stage]
+      prob.f(integrator.du, integrator.u, prob.p, t_stage)
+
+      delta_stage   = alg.delta[stage]
+      gamma1_stage  = alg.gamma1[stage]
+      gamma2_stage  = alg.gamma2[stage]
+      gamma3_stage  = alg.gamma3[stage]
+      beta_stage_dt = alg.beta[stage] * integrator.dt
+      @timeit_debug timer() "Runge-Kutta step" begin
+        Threads.@threads for i in eachindex(integrator.u)
+          integrator.u_tmp2[i] += delta_stage * integrator.u[i]
+          integrator.u[i]       = (gamma1_stage * integrator.u[i] +
+                                   gamma2_stage * integrator.u_tmp2[i] +
+                                   gamma3_stage * integrator.u_tmp3[i] +
+                                   beta_stage_dt * integrator.du[i])
+        end
+      end
+    end
+    integrator.iter += 1
+    integrator.t += integrator.dt
+
+    # handle callbacks
+    if callbacks isa CallbackSet
+      for cb in callbacks.discrete_callbacks
+        if cb.condition(integrator.u, integrator.t, integrator)
+          cb.affect!(integrator)
+        end
+      end
+    end
+  end
+
+  return (t=prob.tspan,
+          u=(copy(prob.u0), copy(integrator.u)),
+          prob=integrator.sol.prob)
+end
+
+# get a cache where the RHS can be stored
+get_du(integrator::SimpleIntegrator3Sstar) = integrator.du
+
+# some algorithms from DiffEq like FSAL-ones need to be informed when a callback has modified u
+u_modified!(integrator::SimpleIntegrator3Sstar, ::Bool) = false
+
+# used by adaptive timestepping algorithms in DiffEq
+function set_proposed_dt!(integrator::SimpleIntegrator3Sstar, dt)
+  integrator.dt = dt
+end
+
+# stop the time integration
+function terminate!(integrator::SimpleIntegrator3Sstar)
+  integrator.finalstep = true
+  empty!(integrator.opts.tstops)
+end
+
+# used for AMR
+function Base.resize!(integrator::SimpleIntegrator3Sstar, new_size)
+  resize!(integrator.u, new_size)
+  resize!(integrator.du, new_size)
+  resize!(integrator.u_tmp, new_size)
```

Comment:
You need to `resize!` all temporary storage elements, e.g. `u_tmp2`.

Reply:
Oh weh, that would have been fun to debug for AMR cases... Thanks for the thorough review! Fixed in 5af2593.